impl/spanner/directory/directory.go

```diff
@@ -0,0 +1,232 @@
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package directory reads and writes Key Transparency Directory information.
+package directory
+
+import (
+	"context"
+	"time"
+
+	"cloud.google.com/go/spanner"
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/google/keytransparency/core/directory"
+	"github.com/google/trillian/crypto/keyspb"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	ktspanner "github.com/google/keytransparency/impl/spanner"
+	tpb "github.com/google/trillian"
+)
+
+const table = "Directories"
+const createTable = `CREATE TABLE Directories (
```

Comment:
This duplicates the [definition](https://github.com/google/keytransparency/blob/468c23f9bd6d82b4bab3d998cffc33f9b093acb4/impl/spanner/keytransparency.ddl#L8) in the DDL file. How are they connected? What if they are out of sync?

Reply:
I'm trying to make up my mind about the canonical work flow here. 
- Users need to setup databases with the proper schema
- `directory.New` calls update database schema
- This could also load the DDL file directly.

Reply:
This this a list of options, or the model you came up with? I think that setting up a database as a distinct step has some benefits:

- Schema definition is not hidden in bits of code. Easier to find it and read (bonus: nice formatting of DDL and possibility of static checks on it).
- Easier (for clients) to discover schema updates, and take the corresponding action to update the DB instance.

Reply:
It might be beneficial to have multiple DDL files / steps, if you have some fairly independent parts of the pipeline.

Reply:
Removed `creatTable` in favor of `ktspanner.ReadDDL()`